i3lock.c

```diff
@@ -1006,6 +1010,9 @@ int main(int argc, char *argv[]) {
     if (main_loop == NULL)
         errx(EXIT_FAILURE, "Could not initialize libev. Bad LIBEV_FLAGS?\n");
 
+    pam_state = STATE_PAM_IDLE;
+    START_TIMER(clear_pam_lock_timeout, TSTAMP_N_SECS(0.25), redraw_timeout);
```

Comment:
You're right, that timer is useless, it could be a simple call to `redraw_screen()`. That call should be needed since it can take some time before successfully grabbing the pointer and therefore we would need to clear the subsequent "locking..." message.